### DISCUSSION TITLE
Gestures

### DISCUSSION THREAD

**Discussion by TopperDEL** [StereoKit/StereoKit#111] [Q&A] [answered]
How are Gestures implemented? Are you using some OpenXR-stuff for this? What if I wanted a specific gesture to be recognized. More specifically: would it be possible to recognize a "flip" with the fingers?

**Comment by maluoi** [comment, marked as answer]
Gestures are an interesting topic! At Microsoft, our general advice is to avoid gestures, since they need to be taught, and teaching gestures is _very_ hard. I've spent a lot of time giving HoloLens demos, and even simple stuff like 'air tap' can be monumentally difficult for many people. It's much better to do 'natural interactions'. A natural interaction might be a 'gesture' depending on how you define it anyhow, but the mentality you use to approach it is key.

Most interactions like this can be done with small pieces of math! A distance check between finger joints, or a dot product on the palm direction, the simpler it is, the more reliable it is to detect or perform. [This document](https://stereokit.net/Pages/Guides/Using-Hands.html) is the official hands docs, and shows a hand facing interaction. [This one is a little more gesture-like](https://twitter.com/koujaku/status/1226333798926651393) and is [accomplished](https://github.com/maluoi/StereoKit/blob/master/StereoKit/Framework/HandMenu/HandMenuRadial.cs) entirely by checking the distance between the tip of a finger, and the root of the finger.

But yeah, try and avoid secret handshake stuff. I don't know too much about the topic myself, I've seen some statistics? based code, and some machine learning code, but I wouldn't want to use either.

**Reply by TopperDEL** [reply]
Thanks for the in-depth answer! I do not plan to have much Gestures - but for that specific app a "flip" would fit so nicely. Let's see - I'll have a look at the resources you've mentioned, too!

**Reply by TopperDEL** [reply]
I found a way to code my own gesture and it was easier than thought thanks to the amazing basic you provide (like the positions of the finger-tips and such).
